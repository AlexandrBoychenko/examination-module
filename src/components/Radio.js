import React from 'react';
import { setLocalStorage } from '../helpers'

class Radio extends React.Component {
    constructor(props) {
        super(props);
        this.handleChangeRadio = this.handleChangeRadio.bind(this);
    }

    componentDidMount() {
        setLocalStorage(this.props.id, this.props.right);
    }

    handleChangeRadio(e) {
        this.props.context.onAnswerChangeRadio(e.target.value, this.props.id);
    }
    
    renderItems() {
        return this.props.items.map((item) => {
            return this.renderInput(item)
        });
    }

    renderInput(item) {
        this.className = 'radio-list';
        return (
            <li key={item} >
                <label className="container">{item}
                    <input
                        type="radio"
                        name={this.props.id}
                        value={item}
                        onChange={this.handleChangeRadio}
                    />
                    <span className="check-mark"></span>
                </label>
            </li>
        );            
    }

    render() {
        return (
            <ul className={this.className}>
                {this.renderItems()}
            </ul>
        )
    }
}

export default Radio;
