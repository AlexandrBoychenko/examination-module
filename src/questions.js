export default [
    {
        id: "answer01",
        value: 'Сколько планет в солнечной системе?',
        items: [7, 5, 9, 8],
        right: 8,
        type: "radio",
        component: "AnswerInput",
        context: this
    },
    {
        id: "answer02",
        value: 'Что характерно для кометы?',
        items: [
            "Имеет хвост",
            "Вращается вокруг солнца",
            "Состоит из газа и пыли",
            "Существует только во внутренней области солнечной системы",
            "Не имеет ядра"
        ],
        right: [
            "Имеет хвост",
            "Вращается вокруг солнца",
        ],
        type: "checkbox",
        component: "AnswerInput",
        context: this
    },
    {
        id: "answer03",
        value: 'Какая из планет солнечной системы имеет наибольший объем?',
        items: null,
        right: 'Юпитер',
        type: null,
        component: "AnswerTextInput",
        context: this
    },
    {
        id: "answer04",
        value: 'Как называется естественный спутник Земли?',
        items: [
            "Ганимед",
            "Луна",
            "Фобос",
            "Титан",
            "Европа"
        ],
        right: "Луна",
        type: null,
        component: "AnswerSelect",
        context: this
    },
    {
        id: "answer05",
        value: 'Достигнул ли к настоящему моменту космический аппарат Voyager-2, сконструированный "NASA",' +
        ' пределов солнечной системы?',
        items: ["Да", "Нет"],
        right: "Да",
        type: "radio",
        component: "AnswerInput",
        context: this
    }
];
