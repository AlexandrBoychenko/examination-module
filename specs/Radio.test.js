import React from 'react';
import Radio from '../src/Components/Radio';
import renderer from 'react-test-renderer';
import questions from '../src/questions';
import { shallow, mount } from 'enzyme';


/*test('It should have all properties', () => {
    questions.forEach((question) => {
        if (question.type === 'Radio') {
            expect(question).toMatchSnapshot({
                id: expect.any(String),
                value: expect.any(String),
                items: expect.any(Array),
                right: expect.anything(),
                type: expect.stringMatching('Radio')
            });
        }
    })
});*/

test("check the onChange callback", () => {
    const wrapper = mount(<Radio {...testRadioProps} />);

    const RadioInputComponent = wrapper.find("input").first();
    RadioInputComponent.simulate("change", {target: {
        value: '7'}});
    expect(RadioInputComponent.props().checked).toBe(true);
});

const testRadioProps = {
    id: "question01",
    items: [7, 5, 9, 8],
    right: 8,
    context: {onAnswerChangeRadio: () => {}},
    pastValues: {},
    parentState: {}
};