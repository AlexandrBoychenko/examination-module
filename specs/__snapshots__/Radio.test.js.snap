// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should be render correctly 1`] = `
<ul>
  <li>
    <label
      className="container"
    >
      7
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={7}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      5
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      9
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={9}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      8
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={8}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
</ul>
`;

exports[`Radio It should be render correctly 1`] = `
<ul>
  <li>
    <label
      className="container"
    >
      7
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={7}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      5
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      9
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={9}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      8
      <input
        checked={false}
        name="question01"
        onChange={[Function]}
        type="radio"
        value={8}
      />
      <span
        className="check-mark"
      />
    </label>
  </li>
</ul>
`;
